homestead.root/usr/share/clearwater/cassandra-schemas/homestead_cache.sh

```diff
@@ -1,50 +1,26 @@
 #! /bin/bash
 
-. /etc/clearwater/config
-if [ ! -z $signaling_namespace ]
-then
-  if [ $EUID -ne 0 ]
-  then
-    echo "When using multiple networks, schema creation must be run as root"
-    exit 2
-  fi
-  namespace_prefix="ip netns exec $signaling_namespace"
-fi
-
 if [[ ! -e /var/lib/cassandra/data/homestead_cache ]];
 then
-  header="Waiting for Cassandra"
-  let "cnt=0"
-  $namespace_prefix netstat -na | grep -q ":7199[^0-9]"
+  # Wait for the cassandra cluster to come online
+  count=0
+  /usr/share/clearwater/bin/poll_cassandra.sh --no-grace-period
+
   while [ $? -ne 0 ]; do
-    sleep 1
-    printf "${header}."
-    header=""
-    let "cnt=$cnt + 1"
-    if [ $cnt -gt 120 ]; then
-      printf "*** ERROR: Cassandra did not come online!\n"
+    ((count++))
+    if [ $count -gt 120 ]; then
+      echo "Cassandra isn't responsive, unable to add schemas"
       exit 1
     fi
-    $namespace_prefix netstat -na | grep -q ":7199[^0-9]"
-  done
-  let "cnt=0"
-  $namespace_prefix netstat -na | grep "LISTEN" | awk '{ print $4 }' | grep -q ":9160\$"
-  while [ $? -ne 0 ]; do
+
     sleep 1
-    printf "${header}+"
-    header=""
-    let "cnt=$cnt + 1"
-    if [ $cnt -gt 120 ]; then
-      printf "*** ERROR: Cassandra did not come online!\n"
-      exit 1
-    fi
-    $namespace_prefix netstat -na | grep "LISTEN" | awk '{ print $4 }' | grep -q ":9160\$"
+    /usr/share/clearwater/bin/poll_cassandra.sh --no-grace-period
   done
 
   echo "CREATE KEYSPACE homestead_cache WITH REPLICATION =  {'class': 'SimpleStrategy', 'replication_factor': 2};
         USE homestead_cache;
         CREATE TABLE impi (private_id text PRIMARY KEY, digest_ha1 text, digest_realm text, digest_qop text, known_preferred boolean) WITH COMPACT STORAGE AND read_repair_chance = 1.0;
-        CREATE TABLE impu (public_id text PRIMARY KEY, ims_subscription_xml text, is_registered boolean) WITH COMPACT STORAGE AND read_repair_chance = 1.0;" | $namespace_prefix cqlsh
+        CREATE TABLE impu (public_id text PRIMARY KEY, ims_subscription_xml text, is_registered boolean) WITH COMPACT STORAGE AND read_repair_chance = 1.0;" | /usr/share/clearwater/bin/run-in-signaling-namespace cqlsh
```

Comment:
namespace_prefix seems to be used again a few times later on in this script - presumably they need to be fixed up too